lib/membrane_flv_plugin/muxer.ex

```diff
@@ -70,15 +78,37 @@ defmodule Membrane.FLV.Muxer do
   end
 
   @impl true
-  def handle_demand(:output, _size, :buffers, _ctx, state) do
-    # We will request one buffer from the stream that has the lowest timestamp
-    # This will ensure that the output stream has reasonable audio / video balance
-    {pad, _dts} = Enum.min_by(state.last_dts, &elem(&1, 1))
-    {[demand: {pad, 1}], state}
+  def handle_buffer(pad, buffer, _ctx, state) do
+    state.queue
+    |> TimestampQueue.push_buffer_and_pop_available_items(pad, buffer)
+    |> handle_queue_output(state)
   end
 
   @impl true
-  def handle_buffer(Pad.ref(type, stream_id) = pad, buffer, _ctx, state) do
+  def handle_stream_format(pad, format, _ctx, state) do
+    state.queue
+    |> TimestampQueue.push_stream_format(pad, format)
+    |> TimestampQueue.pop_available_items()
```

Comment:
Looking at it now, what's the point of having `push_buffer_and_pop_available_items` while we don't have `push_stream_format_and_pop_available_items` and friends?

Reply:
It is done this way because `push_buffer` returns `{suggested_actions, queue}`, so it cannot be just piped to the `pop_available_buffers`, while `push_stream_format` returns `queue`